modules/integration/tests-integration/src/test/java/org/wso2/mdm/integration/mobileDevice/OracleMobileDeviceManagement.java

```diff
@@ -0,0 +1,157 @@
+        //enroll additional 9 devices
+        enrollDevice(Constants.DEVICE_ID_2, Constants.AndroidEnrollment.DEVICE_TWO_ENROLLMENT_DATA);
+        enrollDevice(Constants.DEVICE_ID_3, Constants.AndroidEnrollment.DEVICE_THREE_ENROLLMENT_DATA);
+        enrollDevice(Constants.DEVICE_ID_4, Constants.AndroidEnrollment.DEVICE_FOUR_ENROLLMENT_DATA);
+        enrollDevice(Constants.DEVICE_ID_5, Constants.AndroidEnrollment.DEVICE_FIVE_ENROLLMENT_DATA);
+        enrollDevice(Constants.DEVICE_ID_6, Constants.AndroidEnrollment.DEVICE_SIX_ENROLLMENT_DATA);
+        enrollDevice(Constants.DEVICE_ID_7, Constants.AndroidEnrollment.DEVICE_SEVEN_ENROLLMENT_DATA);
+        enrollDevice(Constants.DEVICE_ID_8, Constants.AndroidEnrollment.DEVICE_EIGHT_ENROLLMENT_DATA);
+        enrollDevice(Constants.DEVICE_ID_9, Constants.AndroidEnrollment.DEVICE_NINE_ENROLLMENT_DATA);
+        enrollDevice(Constants.DEVICE_ID_10, Constants.AndroidEnrollment.DEVICE_TEN_ENROLLMENT_DATA);
+    }
+
+    private void enrollDevice(String deviceId, String deviceEnrollmentData) throws Exception{
+        JsonObject enrollmentData = PayloadGenerator.getJsonPayload(
+                Constants.AndroidEnrollment.ENROLLMENT_ADDITIONAL_DEVICES_PAYLOAD_FILE_NAME,
+                deviceEnrollmentData);
+        enrollmentData.addProperty(Constants.DEVICE_IDENTIFIER_KEY, deviceId);
+        MDMResponse response = client.post(Constants.AndroidEnrollment.ENROLLMENT_ENDPOINT, enrollmentData.toString());
+        Assert.assertEquals(HttpStatus.SC_OK, response.getStatus());
+
+    }
+
+    @Test(dependsOnMethods = {"addEnrollment"}, description = "Test count devices")
+    public void testCountDevices() throws Exception {
+        MDMResponse response = client.get(Constants.MobileDeviceManagement.GET_DEVICE_COUNT_ENDPOINT);
+        Assert.assertEquals(HttpStatus.SC_OK, response.getStatus());
+        Assert.assertTrue(response.getBody().equals(Constants.MobileDeviceManagement.NO_OF_DEVICES));
+
+    }
+    @Test(dependsOnMethods = {"addEnrollment"}, description = "Test view devices")
+    public void testViewDevices() throws Exception {
+        MDMResponse response = client.get(Constants.MobileDeviceManagement.GET_ALL_DEVICES_ENDPOINT);
+        Assert.assertEquals(HttpStatus.SC_OK, response.getStatus());
+    }
+
+    @Test(dependsOnMethods = {"addEnrollment"}, description = "Test view device types")
+    public void testViewDeviceTypes() throws Exception {
+        MDMResponse response = client.get(Constants.MobileDeviceManagement.VIEW_DEVICE_TYPES_ENDPOINT);
+        Assert.assertEquals(HttpStatus.SC_OK, response.getStatus());
+        Assert.assertEquals(PayloadGenerator.getJsonPayloadToString
+                (Constants.MobileDeviceManagement.VIEW_DEVICE_RESPONSE_PAYLOAD_FILE_NAME),response.getBody());
+        //Response has two device types, because in windows enrollment a windows device is previously enrolled.
+    }
+
+    //Pagination testings for GetAllDevice Function
```

Comment:
Correct method level java docs. And, please add doc comments for other methods too.

Reply:
Fixed in https://github.com/wso2-attic/product-emm/pull/389